### ISSUE TITLE
Inconsistent handling of sequences

### ISSUE THREAD

**Issue by opqdonut** [clj-pdf/clj-pdf#185]
## Empty lists don't work

This works:
```clojure
(pdf [{}
      [:heading "bing"]
      (list [:paragraph "1"]
            [:paragraph "2"])]
     "/tmp/example.pdf")
```
This fails with a NullPointerException:
```clojure
(pdf [{}
      [:heading "bing"]
      (list)]
     "/tmp/example.pdf")
```

## Lists don't work everywhere

This works:
```clojure
(pdf [{}
      [:chapter "chapter"
       [:section "section"
        [:paragraph "1"]
        [:paragraph "2"]]]]
     "/tmp/example.pdf")
```

This fails with "class clojure.lang.PersistentVector cannot be cast to class clojure.lang.Named"
```clojure
(pdf [{}
      [:chapter "chapter"
       [:section "section"
        (list [:paragraph "1"]
              [:paragraph "2"])]]]
     "/tmp/example.pdf")
```

**Comment by opqdonut** [comment]
I ran into the first case with code like
```clojure
[{}
 [:heading "Things"]
 (for [thing things]
   (render-thing thing))
```
a workaround was to change that to
```clojure
[{}
 [:heading "Things"]
 (seq
  (for [thing things]
    (render-thing thing))
```
which works even if `things` is empty

**Comment by yogthos** [comment]
Thanks for the PR, just pushed out `2.4.1` with the fix.

**Comment by opqdonut** [comment]
what do you think about the "Lists don't work everywhere" part of the bug? that's still open

**Comment by yogthos** [comment]
Oh sorry, I'd be in favor of expanding lists/seq into the outer form as Hiccup does. I won't have time to look in the near future though, would you be ok doing a PR for that?

**Comment by opqdonut** [comment]
That might be fun, but let's see if I have time.

**Comment by yogthos** [comment]
sounds like a plan :)

**Comment by opqdonut** [comment]
I made one attempt at a solution, but got stuck on the fact that I'm not sure what the contract of `make-section` is supposed to be. Things like `(defmethod render :section` seem to assume it returns something, but on the other hand the implementation of `make-section` has a `doseq` case where it doesn't return anything.

I think if I proceed with this I'll end up refactoring the rendering logic quite a bit to be able to understand it.

https://github.com/clj-pdf/clj-pdf/compare/master...opqdonut:sequences

**Comment by yogthos** [comment]
The `doseq` version does look wrong to me as well, and it doesn't appear to be used anywhere. I've just pushed an update removing it.

**Comment by opqdonut** [comment]
thanks, that clarified the situation for me and I managed to write a PR. I tried to find all the places that need `flatten-seqs` by grepping for `doseq` :)

**Comment by opqdonut** [comment]
ah there are `doseq`s left outside section/core.clj, I'll tackle those next

**Comment by opqdonut** [comment]
In summary, I think this issue can be closed after #188 is merged.
